File: api.game.liveinteractiveworks.com/guide.php
<?php
header('Content-type: application/octet-stream');

function decrypt($packet)
{
	return openssl_decrypt($packet, 'DES-CBC', 'FrPg-109', OPENSSL_RAW_DATA, 'sX5nJEQL');
}
function encrypt($packet)
{
	return openssl_encrypt(json_encode($packet), 'DES-CBC', 'FrPg-109', OPENSSL_RAW_DATA, 'sX5nJEQL');
}

/*
RES Numbers:
SUCCESS 0;
MAINTENANCE 1;
VERSION_UP 2;
SERVICE_CLOSE 3;
FAILURE_NETWORK 1001;
FAILURE_PARAM 8000;
FAILURE_SESSION 8001;
FAILURE_MULTI_LOGIN 8002;
FAILURE_CONTINUOUS_ACCESS 8003;
FAILURE_SERVER 9000;
FAILURE_READ_FILE 9001;
*/

/*Request Params:
{"pid":-1,"ver":3,"deviceType":2,"terminalId":"Device Unique ID","CC":0}
*/

$postRequest = file_get_contents('php://input');
if(isset($postRequest))
{
	$decryptedData = json_decode(decrypt($postRequest), true);
	
	$packet = array();
	switch($decryptedData['pid'])
	{
		case -1: //P_GET_URL
			$packet = array('res' => 0,
				'api' => 'http://127.0.0.1:2090/gw000.php',
				'nsc' => 'http://127.0.0.1:2090/nativeBridge/native/session.php',
				'mainteUrl' => 'https://www.liveinteractiveworks.com/contents/frpg/shinsubarashiki',
				'termsUrl' => 'https://www.liveinteractiveworks.com/contents/frpg/shinsubarashiki/terms/',
				'privacyUrl' => 'https://www.liveinteractiveworks.com/contents/frpg/shinsubarashiki/privacy/',
				'licenseUrl' => 'https://www.liveinteractiveworks.com/contents/frpg/shinsubarashiki/license',
				'commerceUrl' => 'https://www.liveinteractiveworks.com/contents/frpg/shinsubarashiki/commerce',
				'ticketUrl' => 'http://127.0.0.1:2090/ticket.php',
				'storeUrl' => 'https://play.google.com/store/apps/details?id=com.square_enix.android_googleplay.NTWEWYXFIELDWALKRPG',
				'eventEndDate' => '1937679599',
				//'eventEndDate' => '1637679599',			
			);
		break;
	}
	echo encrypt($packet);
}
?>

File: APIHandlers/guide.go
package APIHandlers

import (
	Consts_Protocol "FRPGServer/Consts/Protocol"
	Consts_RES "FRPGServer/Consts/Res"
	"FRPGServer/Utils"
	db_commands "FRPGServer/db/commands"
	"encoding/json"
	"io"
	"log/slog"
	"net/http"
	"os"
	"strconv"
)

func GuideHandler(w http.ResponseWriter, r *http.Request) {
	slog.Info("Guide Hit")
	RequestBody, err := io.ReadAll(r.Body)
	if err != nil {
		ErrorInfo := Utils.FormatError(err.Error())
		slog.Error("Empty Body",
			"File", ErrorInfo.FileName+":"+strconv.Itoa(ErrorInfo.Line),
			"Function", ErrorInfo.FunctionName,
			"ErrorDetail", ErrorInfo.ErrorText)
	}

	DecryptedBody, err := Utils.DESDecrypt(RequestBody)
	if err != nil {
		ErrorInfo := Utils.FormatError(err.Error())
		slog.Error("Decrypt Failed",
			"File", ErrorInfo.FileName+":"+strconv.Itoa(ErrorInfo.Line),
			"Function", ErrorInfo.FunctionName,
			"ErrorDetail", ErrorInfo.ErrorText)
	}
	slog.Debug(string(DecryptedBody))
	var JSONRequest Generic_Request
	err = json.Unmarshal(DecryptedBody, &JSONRequest)
	if err != nil {
		ErrorInfo := Utils.FormatError(err.Error())
		slog.Error("JSON unmarshal failed",
			"File", ErrorInfo.FileName+":"+strconv.Itoa(ErrorInfo.Line),
			"Function", ErrorInfo.FunctionName,
			"ErrorDetail", ErrorInfo.ErrorText)
	}

	if !db_commands.CheckDeviceIdExists(JSONRequest.TerminalId) {
		db_commands.CreateNewUser(JSONRequest.TerminalId)
	}

	switch JSONRequest.PID {
	case Consts_Protocol.GET_URL:
		NetResultGetUrl(w, r)
	}
}

func NetResultGetUrl(w http.ResponseWriter, r *http.Request) {
	Domain := os.Getenv("DOMAIN")
	Port := os.Getenv("PORT")

	var Response Get_URL_Response
	Response.RES = Consts_RES.SUCCESS
	Response.API = "http://" + Domain + ":" + Port + "/gw000.php"
	Response.NSC = "http://" + Domain + ":" + Port + "/nativeBridge/native/session.php"
	Response.MainteURL = "https://www.liveinteractiveworks.com/contents/frpg/shinsubarashiki"
	Response.TermsURL = "https://www.liveinteractiveworks.com/contents/frpg/shinsubarashiki/terms"
	Response.PrivacyURL = "https://www.liveinteractiveworks.com/contents/frpg/shinsubarashiki/privacy"
	Response.LicenseURL = "https://www.liveinteractiveworks.com/contents/frpg/shinsubarashiki/license"
	Response.CommerceURL = "https://www.liveinteractiveworks.com/contents/frpg/shinsubarashiki/commerce"
	Response.TicketURL = "http://" + Domain + ":" + Port + "/ticket.php"
	Response.StoreURL = "https://play.google.com/store/apps/details?id=com.square_enix.android_googleplay.NTWEWYXFIELDWALKRPG"
	Response.EventEndDate = 1937679599 //Original date: 1637679599

	JSONResponse, err := json.Marshal(Response)
	if err != nil {
		ErrorInfo := Utils.FormatError(err.Error())
		slog.Error("Failed to create json response",
			"File", ErrorInfo.FileName+":"+strconv.Itoa(ErrorInfo.Line),
			"Function", ErrorInfo.FunctionName,
			"ErrorDetail", ErrorInfo.ErrorText)
	}

	sendbyte := Utils.DESEncrypt(JSONResponse)

	w.Header().Set("Content-Type", "application/octet-stream")
	w.Write(sendbyte)
}
